### PR TITLE
#755 Fix problem with Joined

### PR DESCRIPTION
**Pull request by pbenety**
Fix problem with org.cactoos.iterable.Joined for #755

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#789](https://github.com/yegor256/cactoos/issues/789) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#789](https://github.com/yegor256/cactoos/issues/789) is assigned to @sharpler[/z](https://www.0crat.com/u/sharpler), here is [why](https://www.0crat.com/footprint/C63314D6Z/1170230120); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by pbenety** [comment]
@0crat waiting for review #789

File: src/test/java/org/cactoos/iterator/JoinedTest.java [hunk 1]
```diff
@@ -56,4 +56,16 @@ public void joinsIterators() {
         );
     }
 
+    @Test
+    public void callsNextDirectlyOnNonEmptyIterator() {
+        MatcherAssert.assertThat(
+            "Cannot call next method directly on non-empty iterator",
+            new Joined<Integer>(
```
Comment (SharplEr) [inline src/test/java/org/cactoos/iterator/JoinedTest.java]: @pbenety replace with `<>`

File: src/test/java/org/cactoos/iterator/JoinedTest.java [hunk 2]
```diff
@@ -56,4 +56,16 @@ public void joinsIterators() {
         );
     }
```
Comment (SharplEr) [inline src/test/java/org/cactoos/iterator/JoinedTest.java]: @pbenety Please add test which throw `NoSuchElementException` with empty `Joined`.

**Comment by SharplEr** [comment]
@pbenety  Please see my remarks.

**Comment by pbenety** [comment]
@SharplEr done

**Comment by SharplEr** [comment]
@pbenety Please see my final remarks. And again sorry for bad advice about `<>` usage. When you see reviewer could be wrong, it's okay to say him about it.

**Comment by pbenety** [comment]
@SharplEr done

**Comment by SharplEr** [comment]
@llorllale good to merge
@pbenety Cool

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@52d3727c-6bc0-4a6a-9013-adf9f764d0c3, thread: AsyncFlush-3
com.zerocracy.farm.reactive.StkGroovy[125] java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/in/orders/close_job.groovy
sun.reflect.NativeMethodAccessorImpl[-2] java.lang.reflect.InvocationTargetException: null
com.mongodb.connection.InternalStreamConnection[457] com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
java.net.SocketInputStream[-2] java.net.SocketTimeoutException: Read timed out
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm)

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @sharpler[/z](https://www.0crat.com/u/sharpler), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#789](https://github.com/yegor256/cactoos/issues/789) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @sharpler[/z](https://www.0crat.com/u/sharpler)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)
